include/kos/barrier.h

```diff
@@ -0,0 +1,42 @@
+/* KallistiOS ##version##
+
+   kos/barrier.h
+   Copyright (C) 2023 Lawrence Sebald
+*/
+
+#ifndef __KOS_BARRIER_H
+#define __KOS_BARRIER_H
+
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
+/** \file   kos/barrier.h
+    \brief  Definitions thread barriers.
+
+    TODO: Write barrier description.
```

Comment:
```suggestion
    Thread barriers are used to synchronize the progress of multiple threads. A 
    barrier causes threads to wait until a specified number of threads have reached
    a certain execution point, ensuring a consistent state across different 
    execution paths.

    The implementation uses a mutex and a condition variable to manage the 
    synchronization process. Threads increment a waiting counter upon reaching the 
    barrier and sleep on the condition variable if they are not the last expected 
    thread. When the final thread reaches the barrier, it triggers a broadcast on 
    the condition variable to wake up all waiting threads. The barrier can be reset 
    for reuse or destroyed as needed.

    This synchronization primitive is essential for scenarios in parallel 
    programming where tasks executed by multiple threads must reach a certain point 
    before any can proceed, ensuring data consistency and coordination among 
    threads.
```

Reply:
Thanks for writing up that description. I'm gonna go ahead and add that in, but without the second paragraph, as I don't think the user needs to know necessarily how it is implemented internally (and that way we don't have to change that description if the internal implementation were to ever change).